src/corefx/System.Globalization.Native/calendarData.c

```diff
@@ -448,31 +477,36 @@ bool EnumAbbrevEraNames(const char* locale,
     {
         UErrorCode status = U_ZERO_ERROR;
 
-        UResourceBundle* rootResBundle = ures_open(nullptr, localeNamePtr, &status);
-        UResourceBundleHolder rootResBundleHolder(rootResBundle, status);
+        UResourceBundle* rootResBundle = ures_open(NULL, localeNamePtr, &status);
 
-        UResourceBundle* calResBundle = ures_getByKey(rootResBundle, "calendar", nullptr, &status);
-        UResourceBundleHolder calResBundleHolder(calResBundle, status);
+        UResourceBundle* calResBundle = ures_getByKey(rootResBundle, "calendar", NULL, &status);
 
         UResourceBundle* targetCalResBundle =
-            ures_getByKey(calResBundle, GetCalendarName(calendarId), nullptr, &status);
-        UResourceBundleHolder targetCalResBundleHolder(targetCalResBundle, status);
+            ures_getByKey(calResBundle, GetCalendarName(calendarId), NULL, &status);
 
-        UResourceBundle* erasColResBundle = ures_getByKey(targetCalResBundle, "eras", nullptr, &status);
-        UResourceBundleHolder erasColResBundleHolder(erasColResBundle, status);
+        UResourceBundle* erasColResBundle = ures_getByKey(targetCalResBundle, "eras", NULL, &status);
 
-        UResourceBundle* erasResBundle = ures_getByKey(erasColResBundle, "narrow", nullptr, &status);
-        UResourceBundleHolder erasResBundleHolder(erasResBundle, status);
+        UResourceBundle* erasResBundle = ures_getByKey(erasColResBundle, "narrow", NULL, &status);
 
         if (U_SUCCESS(status))
         {
             EnumUResourceBundle(erasResBundle, callback, context);
+            ures_close(rootResBundle);
+            ures_close(calResBundle);
+            ures_close(targetCalResBundle);
+            ures_close(erasColResBundle);
+            ures_close(erasResBundle);
             return true;
         }
 
         // Couldn't find the data we need for this locale, we should fallback.
         if (localeNameBuf[0] == 0x0)
         {
+            ures_close(rootResBundle);
```

Comment:
It is better to do this sort of error handling using `goto done` pattern. It avoids duplicating the cleanup and mistakes that the duplication tend to lead to.